### ISSUE TITLE
Tab completion

### ISSUE THREAD

**Issue by jcward** [ianxm/ihx#11]
Fantastic lib, I just discovered it!

The one thing I miss compared to `irb` (and bash) is tab completion. If you don't know the fields of a class, this can be enormously helpful. e.g. I type:

`String.`

And press tab. In irb this shows the possible completions:

```
String.__id__
String.__send__
String.allocate
String.ancestors
> String.
```
etc, etc

It uses multiple columns and other fanciness that isn't necessary. But listing the possible completions would make this tool even better. And completions in all cases could be tricky, but class-level and instance-level completions (maybe just parse the symbol before the .) would be a huge help (so I don't have to go reference the docs, which is what interactive mode is all about.)

Any idea how much effort would be involved?

**Comment by gene-pavlovsky** [comment]
I second that, tab completion is a must feature for any interpreter.
UNIX tools traditionally use the GNU Readline library for this functionality, perhaps it could be used via externs, or parts of it could be ported to Haxe?

**Comment by ianxm** [comment]
I agree that tab completion is good but I wouldn't say it's a "must." the python shell doesn't have it. I don't want to have a dependency on readline. ihx isn't a unix tool and should work on OSs that might not have readline.

**Comment by gene-pavlovsky** [comment]
Yeah, I knew offering Readline is a big stretch. And I understand it can be a major task to implement from scratch (unless there already are some nice cross-platform libraries for that?).
